python_modules/dagster-graphql/dagster_graphql/implementation/execution/__init__.py

```diff
@@ -91,21 +96,27 @@ def terminate_pipeline_execution(
         terminate_policy == GrapheneTerminateRunPolicy.MARK_AS_CANCELED_IMMEDIATELY
     )
 
-    if not record:
-        assert_permission(graphene_info, Permissions.TERMINATE_PIPELINE_EXECUTION)
-        return GrapheneRunNotFoundError(run_id)
+    try:
```

Comment:
wrap the permissions assertions in a try/except so that we can return an error containing a run ID. 

This also allows executing the `terminateRuns` mutation without exiting on the first permissions error